packages/govuk-frontend/src/govuk/components/select/template.njk

```diff
@@ -47,7 +47,7 @@
       {# Allow selecting by text content (the value for an option when no value attribute is specified) #}
       {% set effectiveValue = item.value | default(item.text) %}
       <option {%- if item.value !== undefined %} value="{{ item.value }}"{% endif %}
-        {{-" selected" if item.selected | default(effectiveValue == params.value if params.value else false, true) }}
+        {{-" selected" if item.selected | default((effectiveValue == params.value and item.selected != false) if params.value else false, true) }}
```

Comment:
When will this trigger? My brain's spinning in circles...

Reply:
It's a weird case where `govukSelect()` can have a _selected_ value but the item itself is _unselected_ 🤯

So we want to keep the falsey value as long as it's actually `false` and not `null` or `""`

https://github.com/alphagov/govuk-frontend/blob/72c543a0447c15ea61040cf2fbf901005fe1a0c2/packages/govuk-frontend/src/govuk/components/select/select.yaml#L350-L358

It's covered by this test case:

https://github.com/alphagov/govuk-frontend/blob/72c543a0447c15ea61040cf2fbf901005fe1a0c2/packages/govuk-frontend/src/govuk/components/select/template.test.js#L101-L106